ui/store/actions.ts

```diff
@@ -3777,6 +3489,29 @@ export function rejectPendingApproval(
   };
 }
 
+/**
+ * Rejects all approvals for the given messages
+ *
+ * @param messageList - The list of messages to reject
+ */
+export function rejectAllMessages(
+  messageList: [],
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return async (dispatch: MetaMaskReduxDispatch) => {
+    return Promise.all(
+      messageList.map(
+        async ({ id }) =>
+          await dispatch(
```

Comment:
Is it safer to avoid multiple dispatches here and not use the `rejectPendingApproval` action but instead submit the requests directly?

Each call to the `rejectPendingApproval` action is currently doing a full state update and trying to close the notification.

Reply:
Make sense, fixed.